packages/hdwallet-keepkey/src/ethereum.ts

```diff
@@ -71,7 +71,9 @@ export async function ethSignTx(transport: KeepKeyTransport, msg: ETHSignTx): Pr
   return transport.lockDuring(async () => {
     const est: EthereumSignTx = new EthereumSignTx();
     est.setAddressNList(msg.addressNList);
-    est.setNonce(arrayify(msg.nonce));
+    let nonce = arrayify(msg.nonce);
+    while (nonce.length > 0 && nonce[0] == 0x00) nonce = nonce.slice(1);
+    est.setNonce(nonce);
```

Comment:
You could also do something like:

```suggestion
    const idx = arrayify(msg.nonce).findIndex((x) => x != 0x00);
    est.setNonce(idx > -1 ? msg.nonce?.slice(idx) : [0x00]);
```
I'm not sure which is stylistically preferable.